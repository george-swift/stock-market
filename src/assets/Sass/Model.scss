.model {
  background-color: #f8f9fa;
  min-height: 85vh;
  padding: 1.5rem 0;
  color: #212529;

  .row {
    width: 100%;
    margin: 0;
    padding-left: 1rem;
    padding-right: 1rem;
  }
}

@media screen and (min-width: 800px) {
  .price {
    text-align: center;
  }
}
