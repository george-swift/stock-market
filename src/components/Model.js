import PropTypes from 'prop-types';

const Model = ({ company }) => {
  const {
    symbol,
    companyName,
    ceo,
    website,
    ipoDate,
    industry,
    sector,
    fullTimeEmployees,
    price,
    changes,
    volAvg,
    mktCap,
    dcf,
    description,
  } = company;

  return (
    <>
      <p className="col-lg-3">
        <span className="fw-bold">CEO:</span>
        <br />
        {ceo}
      </p>
      <p className="col-lg-4">
        <span className="fw-bold">Website:</span>
        <br />
        <a href={website} target="_blank" rel="noreferrer">{website}</a>
      </p>
      <p className="col-lg-3">
        <span className="fw-bold">Company Name:</span>
        <br />
        {companyName}
      </p>
      <p className="col-lg-2">
        <span className="fw-bold">Symbol:</span>
        <br />
        {symbol}
      </p>

      <p className="col-lg-3">
        <span className="fw-bold">Sector:</span>
        <br />
        {sector}
      </p>
      <p className="col-lg-4">
        <span className="fw-bold">Industry:</span>
        <br />
        {industry}
      </p>
      <p className="col-lg-3">
        <span className="fw-bold">Fulltime Employees:</span>
        <br />
        {Number(fullTimeEmployees).toLocaleString()}
      </p>
      <p className="col-lg-2">
        <span className="fw-bold">IPO Date:</span>
        <br />
        {ipoDate}
      </p>

      <p className="col-lg-3">
        <span className="fw-bold">Market Cap:</span>
        <br />
        {`$${mktCap.toLocaleString()}`}
      </p>
      <p className="col-lg-2">
        <span className="fw-bold">Discounted Cash Flow:</span>
        <br />
        {`$${dcf}`}
      </p>
      <p className="col-lg-2 price">
        <span className="fw-bold"> Price:</span>
        <br />
        {`$${price}`}
      </p>
      <p className="col-lg-3">
        <span className="fw-bold">Volume Average:</span>
        <br />
        {volAvg.toLocaleString()}
      </p>
      <p className="col-lg-2">
        <span className="fw-bold">Changes:</span>
        <br />
        {changes}
      </p>

      <p className="col-lg-12">
        <span className="fw-bold">Description:</span>
        <br />
        {description}
      </p>
    </>
  );
};

Model.propTypes = {
  company: PropTypes.shape({
    symbol: PropTypes.string,
    companyName: PropTypes.string,
    ceo: PropTypes.string,
    website: PropTypes.string,
    ipoDate: PropTypes.string,
    industry: PropTypes.string,
    sector: PropTypes.string,
    fullTimeEmployees: PropTypes.string,
    price: PropTypes.number,
    changes: PropTypes.number,
    volAvg: PropTypes.number,
    mktCap: PropTypes.number,
    dcf: PropTypes.number,
    description: PropTypes.string,
  }).isRequired,
};

export default Model;
