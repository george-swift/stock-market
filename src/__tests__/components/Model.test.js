import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import Model from '../../components/Model';

const company = {
  symbol: 'TS',
  companyName: 'Test Co',
  ceo: 'Devs',
  website: 'www.test.com',
  ipoDate: '2021-07-31',
  industry: 'SF',
  sector: 'Tech',
  fullTimeEmployees: '1000',
  price: 10,
  changes: 2,
  volAvg: 1255,
  mktCap: 5000,
  dcf: 8,
  description: 'Test componany for Model',
};

describe('Snapshot of the Model component', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<Model company={company} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Querying the Model component', () => {
  beforeEach(() => render(<Model company={company} />));

  it('should have nodes for company name, ceo and website', () => {
    const companyName = screen.getByText('Company Name:').nextSibling.nextSibling;
    expect(companyName).toHaveTextContent(company.companyName);
    const companyCeo = screen.getByText('CEO:').nextSibling.nextSibling;
    expect(companyCeo).toHaveTextContent(company.ceo);
    const companyWebsite = screen.getByText('Website:').nextSibling.nextSibling;
    expect(companyWebsite).toHaveAttribute('href', company.website);
    expect(companyWebsite).toHaveTextContent(company.website);
  });

  it('should have nodes for company symbol, sector and industry', () => {
    const companySymbol = screen.getByText('Symbol:').nextSibling.nextSibling;
    expect(companySymbol).toHaveTextContent(company.symbol);
    const companySector = screen.getByText('Sector:').nextSibling.nextSibling;
    expect(companySector).toHaveTextContent(company.sector);
    const companyIndustry = screen.getByText('Industry:').nextSibling.nextSibling;
    expect(companyIndustry).toHaveTextContent(company.industry);
  });

  it('should convert large values of company full-time employees, market cap and volume average to locale string ', () => {
    const companyEmployees = screen.getByText('Fulltime Employees:').nextSibling.nextSibling;
    expect(companyEmployees).not.toHaveTextContent(company.fullTimeEmployees);
    expect(companyEmployees).toHaveTextContent('1,000');
    const companyMarketCap = screen.getByText('Market Cap:').nextSibling.nextSibling;
    expect(companyMarketCap).not.toHaveTextContent(company.mktCap);
    expect(companyMarketCap).toHaveTextContent('5,000');
    const companyVolumeAvg = screen.getByText('Volume Average:').nextSibling.nextSibling;
    expect(companyVolumeAvg).not.toHaveTextContent(company.volAvg);
    expect(companyVolumeAvg).toHaveTextContent('1,255');
  });

  it('should have nodes for company IPO date, price, DCF, changes and description', () => {
    const companyIpoDate = screen.getByText('IPO Date:').nextSibling.nextSibling;
    expect(companyIpoDate).toHaveTextContent(company.ipoDate);
    const companyPrice = screen.getByText('Price:').nextSibling.nextSibling;
    expect(companyPrice).toHaveTextContent(company.price);
    const companyDcf = screen.getByText('Discounted Cash Flow:').nextSibling.nextSibling;
    expect(companyDcf).toHaveTextContent(company.dcf);
    const companyDescription = screen.getByText('Description:').nextSibling.nextSibling;
    expect(companyDescription).toHaveTextContent(company.description);
  });
});
