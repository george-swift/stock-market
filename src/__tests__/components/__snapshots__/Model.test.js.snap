// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of the Model component should render correctly 1`] = `
Array [
  <p
    className="col-lg-3"
  >
    <span
      className="fw-bold"
    >
      CEO:
    </span>
    <br />
    Devs
  </p>,
  <p
    className="col-lg-4"
  >
    <span
      className="fw-bold"
    >
      Website:
    </span>
    <br />
    www.test.com
  </p>,
  <p
    className="col-lg-3"
  >
    <span
      className="fw-bold"
    >
      Company Name:
    </span>
    <br />
    Test Co
  </p>,
  <p
    className="col-lg-2"
  >
    <span
      className="fw-bold"
    >
      Symbol:
    </span>
    <br />
    TS
  </p>,
  <p
    className="col-lg-3"
  >
    <span
      className="fw-bold"
    >
      Sector:
    </span>
    <br />
    Tech
  </p>,
  <p
    className="col-lg-4"
  >
    <span
      className="fw-bold"
    >
      Industry:
    </span>
    <br />
    SF
  </p>,
  <p
    className="col-lg-3"
  >
    <span
      className="fw-bold"
    >
      Fulltime Employees:
    </span>
    <br />
    1,000
  </p>,
  <p
    className="col-lg-2"
  >
    <span
      className="fw-bold"
    >
      IPO Date:
    </span>
    <br />
    2021-07-31
  </p>,
  <p
    className="col-lg-3"
  >
    <span
      className="fw-bold"
    >
      Market Cap:
    </span>
    <br />
    $5,000
  </p>,
  <p
    className="col-lg-2"
  >
    <span
      className="fw-bold"
    >
      Discounted Cash Flow:
    </span>
    <br />
    $8
  </p>,
  <p
    className="col-lg-2 price"
  >
    <span
      className="fw-bold"
    >
       Price:
    </span>
    <br />
    $10
  </p>,
  <p
    className="col-lg-3"
  >
    <span
      className="fw-bold"
    >
      Volume Average:
    </span>
    <br />
    1,255
  </p>,
  <p
    className="col-lg-2"
  >
    <span
      className="fw-bold"
    >
      Changes:
    </span>
    <br />
    2
  </p>,
  <p
    className="col-lg-12"
  >
    <span
      className="fw-bold"
    >
      Description:
    </span>
    <br />
    Test componany for Model
  </p>,
]
`;
